File: need4spd/euler_25.py
def fib(n):
    a, b = 0, 1
    for i in range(n):
        a, b = b, a + b
    return a

n = 1
while True:
    r = fib(n)
    
    if len(str(r)) == 1000:
        print(n)
        break
    
    n+=1

File: need4spd/euler_25.pl
use strict;
use warnings;
use bigint;

sub fib {
    my ($n) = @_;
    
    my ($a, $b) = (0, 1);
    
    foreach (0..$n) {
        ($a, $b) = ($b, $a+$b);
    }
    
    return $a;
}


my $n = 1;
while(1) {
    my $r = fib($n);
    my $l = length($r); 
    print "$l \n";
    if ($l == 1000) {
        print "$n";
        last;
    }
    
    $n+=1;
}